shared/Microsoft.Extensions.Buffers.Sources/BufferWriter.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Extensions.Buffers
+{
+    internal ref struct BufferWriter<T> where T: IBufferWriter<byte>
```

Comment:
Where is @benaadams 's change to add Commit?

Reply:
[OutputWriter](https://github.com/dotnet/corefxlab/blob/master/src/System.Buffers.Primitives/System/Buffers/OutputWriter.cs)

Could instead make it `IDisposable` and use `Dispose` instead of `Commit`?

Reply:
```csharp
using (var writer = new BufferWriter(_pipeWriter))
{
    writer.Write(_bytesHttpVersion11);
    var statusBytes = ReasonPhrases.ToStatusBytes(statusCode, reasonPhrase);
    writer.Write(statusBytes);
    responseHeaders.CopyTo(ref writer);
    writer.Write(_bytesEndHeaders);
}
```

Reply:
Can't implement interfaces on a ref struct AFAIK

Reply:
😢

Reply:
And `using` doesn't match on `Dispose` but needs `IDisposable` 😞